opa/pkg/main.go

```diff
@@ -185,6 +192,133 @@ func processCheckTarget(ctx context.Context, check customCheck, checkInstance cu
 	return actionItems, nil
 }
 
+func getInsightsChecks() (clusterCheckModel, error) {
+	var jsonResponse clusterCheckModel
+
+	url := os.Getenv("FAIRWINDS_INSIGHTS_HOST") + "/v0/organizations/" + os.Getenv("FAIRWINDS_ORG") + "/clusters/" + os.Getenv("FAIRWINDS_CLUSTER") +
+		"/data/opa/customChecks"
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return jsonResponse, err
+	}
+	req.Header.Set("Authorization", "Bearer "+os.Getenv("FAIRWINDS_TOKEN"))
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return jsonResponse, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		return jsonResponse, fmt.Errorf("failed to retrieve updated checks with a status code of : %d", resp.StatusCode)
+	}
+	responseBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return jsonResponse, err
+	}
+	err = json.Unmarshal(responseBody, &jsonResponse)
+	if err != nil {
+		return jsonResponse, err
+	}
+	return jsonResponse, nil
+}
+
+func refreshLocalChecks(ctx context.Context, dynamicInterface dynamic.Interface) error {
+
+	thisNamespace := "insights-agent"
+	namespaceBytes, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	if err == nil { //Ignore errors because that means this isn't running in a container
+		thisNamespace = string(namespaceBytes)
+	}
+	checkInstances, err := dynamicInterface.Resource(instanceGvr).Namespace(thisNamespace).List(ctx, metav1.ListOptions{})
```

Comment:
Yes that's intended. That's the location to store the Insights checks in. If this is running in a cluster we can see what namespace we're in. If not we default to insights-agent.

It needs to store the checks somewhere and the agent namespace feels the appropriate place. I don't know if it's ever valuable but I left it open that someone could create their own checks outside of the insights-agent namespace and this will never delete those. All of the checks currently apply to every namespace (I think I left a TODO that is something that should change).

Reply:
OK that makes sense to me.